src/Sextant.UWP/NavigationView.xaml.cs

```diff
@@ -0,0 +1,456 @@
+        /// </summary>
+        public bool ModalVisible
+        {
+            get
+            {
+                var popups = VisualTreeHelper.GetOpenPopups(Window.Current);
+                return popups.FirstOrDefault(x => x.Child is ContentDialog) != null;
+            }
+        }
+
+        /// <inheritdoc />
+        public IScheduler MainThreadScheduler => _mainScheduler;
+
+        /// <inheritdoc />
+        public IObservable<IViewModel> PagePopped { get; }
+
+        /// <summary>
+        /// Gets combined backrequested observable from system, backbutton, and xbox controller sources.
+        /// </summary>
+        public IObservable<Unit> BackRequested => Observable.Merge(
+            Observable
+                .FromEventPattern<BackRequestedEventArgs>(
+                x => SystemNavigationManager.GetForCurrentView().BackRequested += x,
+                x => SystemNavigationManager.GetForCurrentView().BackRequested -= x)
+                .Do(ev =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        PopPage(true);
+                        ev.EventArgs.Handled = true;
+                    }
+                })
+                .ToSignal(),
+            Observable
+                .FromEvent<PointerEventHandler, PointerRoutedEventArgs>(
+                handler =>
+                {
+                    PointerEventHandler pointerHandler = (sender, e) => handler(e);
+                    return pointerHandler;
+                },
+                x => PointerPressed += x,
+                x => PointerPressed -= x)
+                .Where(args => args.GetCurrentPoint(this as UIElement).Properties.PointerUpdateKind == Windows.UI.Input.PointerUpdateKind.XButton1Pressed)
+                .Do(args =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        PopPage(true);
+                        args.Handled = true;
+                    }
+                })
+                .ToSignal(),
+            Observable
+                .FromEvent<RoutedEventHandler, RoutedEventArgs>(
+                handler =>
+                {
+                    RoutedEventHandler routedHandler = (sender, e) => handler(e);
+                    return routedHandler;
+                },
+                x => backButton.Click += x,
+                x => backButton.Click -= x)
+                .Do(args =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        PopPage(true);
+                    }
+                })
+                .ToSignal());
+
+        /// <inheritdoc />
+        public IObservable<Unit> PopModal()
+        {
+            if (_contentDialog != null)
+            {
+                _contentDialog.Hide();
+                var viewStackService = Locator.Current.GetService<IViewStackService>();
```

Comment:
UWP does not have a concept of a Navigation Stack?  It doesn't keep a list of everything that has been presented over it's lifecycle?  This feels bad.  Primarily because I am not sure we can guarantee that it will come from the service registered for the `IViewStackService`.

Reply:
trying something new...

Reply:
I did remove all references to `IViewStackService`.